### DISCUSSION TITLE
Connecting ESP32 board with Waveshare touch display with ILI9488 driver - wrong screen resolution

### DISCUSSION THREAD

**Discussion by Rongit53** [Bodmer/TFT_eSPI#2889] [New processor/display support or capability request]
I use the TFT_eSPI library from Bodmer (great job -my respect!!) to connect the Waveshare touch display with ILI9488 driver, using SPI bus. 
I uncommending the line #define ILI9488_DRIVER in the User_Setup.h file, and
the line #include <User_Setups/Setup21_ILI9488.h> in the User_Setup_Select.h file.
I also double checked the pin-connections. After all, no of the TFT_eSPI/ 480x320 examples were running.
After adding the line #define RPI_DISPLAY_TYPE in the Setup21_ILI9488.h file i got it running but with wrong colors.
This issue I could solve with #define TFT_INVERSION_ON.
The remaining problem I have is that the examples are running with a screen resolutin of 240x240.
My question: How I can adjust the screen resolution to 480 x 320?

**Comment by Bodmer** [comment]
Run the Read_User_setup sketch and check the output in the serial moitor window. The displays size, sriver pins etc all should be as you expect. If not then the User_Setup_Select file may be calling up the wrong header, or two headers.

**Reply by Rongit53** [reply]
Thanks for yor answer. In the User_Setup_Select file I just uncommend the header #include <User_Setups/Setup21_ILI9488.h>.
I run the Read_User_setup sketch. That's the output:
Prozessor     = ESP32
Frequency    = 240MHz
Transactions = Yes
Interface    = SPI
Display driver = 9488
Display width  = 320
Display height = 480

MOSI    = GPIO 23
MISO    = GPIO 19
SCK     = GPIO 18
TFT_CS   = GPIO 5
TFT_DC   = GPIO 16
TFT_RST  = GPIO 17

TFT_BL           = GPIO 13
TFT_BACKLIGHT_ON = HIGH

Font GLCD   loaded
Font 2      loaded
Font 4      loaded
Font 6      loaded
Font 7      loaded
Font 8      loaded
Smooth font enabled

Display SPI frequency = 60.00
[/code]
Ln 1, Col 1
DOIT ESP32 DEVKIT V1
on COM5
TFT_eSPI ver = 2.5.0

**Reply by Bodmer** [reply]
Reduce the SPI frequency to 20MHz. If that works, increase to find the upper limit.

**Comment by Rongit53** [comment]
The SPI frequency caused the problem. I can not go higher than 20 MHz.
Now the display is working correctly. Thanks a lot.

**Reply by Bodmer** [reply]
Usually 27MHz is the maximum with these displays.

**Reply by Rongit53** [reply]
I tried 27 MHz but it's to high. For my application I can live with 20 Mhz.
Just fo my understanding , what effects that define:   #define RPI_DISPLAY_TYPE?

**Reply by Bodmer** [reply]
Some boards have serial to parallel circuits, these require 16 bit commands to be sent instead of the normal SPI 8 bit commands. RPI_DISPLAY_TYPE invokes that interface mode. This interface type was first seen on displays intended for the Raspberry Pi, hence the naming.

**Reply by Rongit53** [reply]
Once again my personal thanks to you !!. I think, we can close the discussion?
